Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -1810,6 +1810,7 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     void willStartUserTriggeredZooming();
     void didEndUserTriggeredZooming();
     bool mainFramePluginHandlesPageScaleGesture() const { return m_mainFramePluginHandlesPageScaleGesture; }
+    bool hasMainFramePlugin() const { return m_pluginMaxZoomFactor.has_value() || m_pluginMinZoomFactor.has_value(); }
```

Comment:
Hm... it seems a bit unfortunate to tie `hasMainFramePlugin()` to zooming state, though it seems correct in this context. Maybe we could clarify this by renaming `hasMainFramePlugin()` to something more specific, like `mainFramePluginOverridesViewScale()`, and do:

```
    if (_page->mainFramePluginOverridesViewScale()) {
        if (_page->layoutSizeScaleFactorFromClient() != 1)
            _page->setViewportConfigurationViewLayoutSize(_page->viewLayoutSize(), 1, _page->minimumEffectiveDeviceWidth());
        return;
    }
```

…

```
bool WebPageProxy::mainFramePluginOverridesViewScale() const
{
    if (m_mainFramePluginHandlesPageScaleGesture)
        return false;
    return m_pluginMaxZoomFactor.has_value() || m_pluginMinZoomFactor.has_value();
}
```

Reply:
Yes, I agree, it seems weird for `hasMainFramePlugin()` to show up in zooming logic.

Reply:
I will go with your suggestion

Reply:
Changed in https://github.com/WebKit/WebKit/commit/c160b87e0368d41a8b243807185c6b17e2a2d4a5